core/app/models/spree/property.rb

```
@@ -50,6 +50,21 @@ def uniq_values(product_properties_scope: nil)
+    # @return [String] eg. 'Spree::Metafields::ShortText'
+    def kind_to_metafield_type
+      case kind
+      when 'short_text'
+        'Spree::Metafields::ShortText'
+      when 'long_text'
+        'Spree::Metafields::LongText'
+      when 'number'
+        'Spree::Metafields::Number'
+      when 'rich_text'
+        'Spree::Metafields::RichText'
+      end
```

RuboCop (1.81.1)

[convention] Consider replacing `case-when` with a hash lookup.

(Style/HashLikeCase)